sysbuild/bootloader/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Include MCUboot if enabled.
+if(SB_CONFIG_BOOTLOADER_MCUBOOT)
+  ExternalZephyrProject_Add(
+    APPLICATION mcuboot
+    SOURCE_DIR ${ZEPHYR_MCUBOOT_MODULE_DIR}/boot/zephyr/
+  )
+  # MCUBoot should be first in list, as it is a bootloader.
+  set(IMAGES "mcuboot" ${IMAGES} PARENT_SCOPE)
```

Comment:
I don't see any information as to why MCUBoot being first in the list is needed or desirable.
Or any information about this variable in general (in the doc PR as well).

Reply:
the main reason for this is because MCUBoot default will do a mass erase of the chip:
https://github.com/zephyrproject-rtos/mcuboot/blob/e7415b555134e671ba04f036b02f55cbc087d0e9/boot/zephyr/Kconfig#L449-L451
https://github.com/zephyrproject-rtos/mcuboot/blob/e7415b555134e671ba04f036b02f55cbc087d0e9/boot/zephyr/CMakeLists.txt#L36-L38

Meaning that if we flash MCUBoot after the application, then the application gets wiped :disappointed: 

So per se it's not because it's a bootloader, but because it's MCUBoot and it does a mass erase per default.
Will adjust the comment.

Reply:
@jfischer-no I agree that this is bad behavior from MCUboot, but it is for MCUboot to decide if it should change.

but mentioning it in #43846 is for sure a valid request, so will do.

Reply:
@JordanYates updated comment and added a note on this in the docs.